File: FSNavigationController.m
//
//  FSNavigationController.m
//  FSNavigationController
//
//  Created by Danyun Liu on 3/1/16.
//  Copyright © 2016 liudanyun@gmail.com. All rights reserved.
//


#import "FSNavigationController.h"


@implementation FSNavigationController {
    UIPanGestureRecognizer * _fullScreenPanGestureRecognizer;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wundeclared-selector"
    
    id _cachedInteractionController = [self valueForKey:@"_cachedInteractionController"];
    if (_cachedInteractionController && [_cachedInteractionController respondsToSelector:@selector(handleNavigationTransition:)]) {
        _fullScreenPanGestureRecognizer = [[UIPanGestureRecognizer alloc]initWithTarget:_cachedInteractionController
                                                                                 action:@selector(handleNavigationTransition:)];
        [self.view addGestureRecognizer:_fullScreenPanGestureRecognizer];
    }

#pragma clang diagnostic pop
    
}


@end

File: FSNavigationController.swift
//
//  FSNavigationController.swift
//  swiftexample
//
//  Created by Danyun Liu on 3/2/16.
//  Copyright © 2016 liudanyun.com. All rights reserved.
//

import UIKit

public class FSNavigationController: UINavigationController {
    
    private var _fullScreenPanGestureRecognizer : UIPanGestureRecognizer!

    public override func viewDidLoad() {
        super.viewDidLoad()
        let target = self.valueForKey("_cachedInteractionController")
        if let canRespondsToSelector = target?.respondsToSelector("handleNavigationTransition:") {
            if (canRespondsToSelector) {
                _fullScreenPanGestureRecognizer = UIPanGestureRecognizer(target: target!, action: "handleNavigationTransition:")
                self.view.addGestureRecognizer(_fullScreenPanGestureRecognizer)
            }
        }
    }
}
